### ISSUE TITLE
Resource not accessible by integration

### ISSUE THREAD

**Issue by MarcHagen** [kachick/wait-other-jobs#598]
Since the 2.0.0 it failed to check jobs in our private repos
We use it as a reusable workflow file, so maybe that is related?
But it worked with the v1.3.0, so i suspect the GraphQL to break something here.

It could be that the `secrets.GITHUB_TOKEN` in the reusable workflow does not have the right permissions to make these kinds of calls?

```
Run kachick/wait-other-jobs@6311df9bc7[2](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:2)5c0a5092e2db[3](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:3)22c7cbe19165d[4](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:4)7d
  with:
    github-token: ***
    wait-seconds-before-first-polling: 10
    min-interval-seconds: 1[5](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:5)
    early-exit: true
    retry-method: equal_intervals
    attempt-limits: 1000
    wait-list: []
    skip-list: []
    dry-run: false
Parameters
  {
    "triggeredCommitSha": "711f731488[6](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:6)b83a0c51[7](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:7)92c2dd3c9ac5b3444[8](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:8)6a",
    "runId": 63[9](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:9)0931405,
    "repositoryInfo": {
      "owner": "company",
      "repo": "private-repo"
    },
    "waitSecondsBeforeFirstPolling": [10](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:10),
    "minIntervalSeconds": 15,
    "retryMethod": "equal_intervals",
    "attemptLimits": 1000,
    "isEarlyExit": true,
    "isDryRun": false,
    "waitList": [],
    "skipList": []
  }
Wait about 10 seconds before first polling.
Polling 1: 2023-10-03T09:07:17.981Z
  file:///home/runner/_work/_actions/kachick/wait-other-jobs/63[11](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:11)df9bc725c0a5092e2db322c7cbe19165d47d/dist/index.js:8399
            throw new GraphqlResponseError(
                  ^
  
  GraphqlResponseError: Request failed due to following response errors:
   - Resource not accessible by integration
      at file:///home/runner/_work/_actions/kachick/wait-other-jobs/6311df9bc725c0a5092e2db322c7cbe19165d47d/dist/index.js:8399:17
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Object.next (file:///home/runner/_work/_actions/kachick/wait-other-jobs/6311df9bc725c0a5092e2db322c7cbe19165d47d/dist/index.js:8787:32)
      at async Function.paginate (file:///home/runner/_work/_actions/kachick/wait-other-jobs/6311df9bc725c0a5092e2db322c7cbe19165d47d/dist/index.js:8835:26)
      at async getCheckRunSummaries (file:///home/runner/_work/_actions/kachick/wait-other-jobs/6311df9bc725c0a5092e2db322c7cbe19165d47d/dist/index.js:[12](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:13)779:55)
      at async fetchOtherRunStatus (file:///home/runner/_work/_actions/kachick/wait-other-jobs/6311df9bc725c0a5092e2db322c7cbe19165d47d/dist/index.js:12870:27)
      at async run (file:///home/runner/_work/_actions/kachick/wait-other-jobs/6311df9bc725c0a5092e2db322c7cbe19165d47d/dist/index.js:[13](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:14)033:20) {
    request: {
      query: '\n' +
        '    query GetCheckRuns($owner: String!, $repo: String!, $commitSha: String!, $cursor: String) {\n' +
        '      repository(owner: $owner, name: $repo) {\n' +
        '        object(expression: $commitSha) {\n' +
        '          ... on Commit {\n' +
        '            checkSuites(first: 100, after: $cursor) {\n' +
        '              totalCount\n' +
        '              pageInfo {\n' +
        '                hasNextPage\n' +
        '                endCursor\n' +
        '              }\n' +
        '              nodes {\n' +
        '                status\n' +
        '                conclusion\n' +
        '                workflowRun {\n' +
        '                  databaseId\n' +
        '                  workflow {\n' +
        '                    name\n' +
        '                    resourcePath\n' +
        '                  }\n' +
        '                }\n' +
        '                checkRuns(first: 100) {\n' +
        '                  totalCount\n' +
        '                  pageInfo {\n' +
        '                    hasNextPage\n' +
        '                    endCursor\n' +
        '                  }\n' +
        '                  nodes {\n' +
        '                    databaseId\n' +
        '                    name\n' +
        '                    status\n' +
        '                    detailsUrl\n' +
        '                    conclusion\n' +
        '                  }\n' +
        '                }\n' +
        '              }\n' +
        '            }\n' +
        '          }\n' +
        '        }\n' +
        '      }\n' +
        '    }\n' +
        '  ',
      variables: {
        owner: 'company',
        repo: 'private-repo',
        commitSha: '711f73[14](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:15)886b83a0c51792c2dd3c9ac5b344486a'
      }
    },
    headers: {
      'access-control-allow-origin': '*',
      'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
      'content-encoding': 'gzip',
      'content-security-policy': "default-src 'none'",
      'content-type': 'application/json; charset=utf-8',
      date: 'Tue, 03 Oct 2023 09:07:18 GMT',
      'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
      server: 'GitHub.com',
      'strict-transport-security': 'max-age=3[15](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:16)36000; includeSubdomains; preload',
      'transfer-encoding': 'chunked',
      vary: 'Accept-Encoding, Accept, X-Requested-With',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'deny',
      'x-github-media-type': 'github.v3; format=json',
      'x-github-request-id': '2825:B91C:47A3F3F:4850AA2:651BD9C6',
      'x-ratelimit-limit': '1000',
      'x-ratelimit-remaining': '998',
      'x-ratelimit-reset': '[16](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:17)96327130',
      'x-ratelimit-resource': 'graphql',
      'x-ratelimit-used': '2',
      'x-xss-protection': '0'
    },
    response: {
      data: { repository: { object: null } },
      errors: [
        {
          type: 'FORBIDDEN',
          path: [ 'repository', 'object' ],
          extensions: { saml_failure: false },
          locations: [ { line: 4, column: 9 } ],
          message: 'Resource not accessible by integration'
        }
      ]
    },
    errors: [
      {
        type: 'FORBIDDEN',
        path: [ 'repository', 'object' ],
        extensions: { saml_failure: false },
        locations: [ { line: 4, column: 9 } ],
        message: 'Resource not accessible by integration'
      }
    ],
    data: { repository: { object: null } }
  }
  
  Node.js v[20](https://github.com/company/private-repo/actions/runs/6390931405/job/17345186620?pr=82#step:2:21).5.0
```

Part of the reusable workflow: (company/github-workflows/.github/workflows/dependabot-automerge.yml)
```yaml
name: 🤖 Dependabot Auto Merge
on:
  workflow_call:
    inputs:
      merge-option:
        description: The merge method to use. `merge`, `rebase` or `squash`.
        type: string
        required: false
        default: squash
    secrets:
      # We need a token to enable auto-merge and auto-approve.
      # This token CANNOT be the default `GITHUB_TOKEN` on auto-merge action, otherwise the merge of the PR will not trigger a build (!).
      bot_app_id:
        description: GitHub App ID
        required: true
      bot_app_private_key:
        description: Private RSA key from the GitHub App
        required: true

permissions: {}


jobs:
  wait-on-checks:
    runs-on: novo-k3-nl-dind
    if: ${{ github.event.pull_request.user.login == 'dependabot[bot]' }}
    permissions:
      # wait-on-check requires only checks read
      checks: read
      actions: read
    steps:
      - uses: kachick/wait-other-jobs@6311df9bc725c0a5092e2db322c7cbe19165d47d # v2.0.0
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}

<snipped more jobs depending on wait-on-checks>
```

Repo workflow calling the reusable workflow: (company/private-repo/.github/workflows/dependabot-automerge.yml
```yaml
---
name: 🤖 Dependabot Auto Merge
on:
  pull_request_target:
    types:
      # Dependabot will create the PR
      - opened
      # Dependabot will label the PR
      - labeled
      # Dependabot has rebased the PR
      - synchronize
    branches:
      - main

permissions:
  contents: write # Needed for merging pull requests
  pull-requests: write # Needed to auto-approve and auto-merge pull requests
  checks: read # Needed for "wait-on-checks"
  actions: read # Needed for "wait-on-checks"

jobs:
  dependabot-auto-merge:
    uses: company/github-workflows/.github/workflows/dependabot-automerge.yml@main
    secrets:
      bot_app_id: ${{ secrets.APP_ID }}
      bot_app_private_key: ${{ secrets.APP_PRIVATE_KEY }}
```

**Comment by kachick** [comment]
Thanks for your reporting! I have tested v2 in my private repository and I can found same error when enabled only 2 permissions like public repository.

```yaml
permissions:
  contents: write
  pull-requests: write
```

```
  GraphqlResponseError: Request failed due to following response errors:
   - Resource not accessible by integration
```

However enabling other 2 permissions made it okay

```yaml
permissions:
  contents: write
  pull-requests: write
  checks: read # For private repositories
  actions: read # For private repositories
```

As I understand it, this is same restriction since v1. I noted them as followings.

https://github.com/kachick/wait-other-jobs/blob/9c36ea8bfafcaf8ee7c93b5cad729fb55ba9c1ca/README.md#L135-L142
https://github.com/kachick/wait-other-jobs/blob/9c36ea8bfafcaf8ee7c93b5cad729fb55ba9c1ca/README.md#L84-L88

I have no experience to use this action by reusable workflow in private repository, but I guess this problem isn't a spec changes at least documented scopes🤔

**Comment by kachick** [comment]
I want to try similar reusable workflows in private repository.
By the way, above shared `Part of the reusable workflow: (company/github-workflows/.github/workflows/dependabot-automerge.yml)`  looks like having too small permissions, I think

```
    permissions:
      # wait-on-check requires only checks read
      checks: read
      actions: read
```

**Comment by MarcHagen** [comment]
Ill see if the extra `contents: write`, hoping `read` would be enough, from a security perspective.
It seems it was not needed by the "old" api, but GraphQL is different... (jeej)

**Comment by kachick** [comment]
Keeping minimum required permissions should be happy also for me. :pray: 
Especially this action is not using mutations.

Reading about some articles... 

https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

```
When a workflow is triggered by the [pull_request_target](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) event, the GITHUB_TOKEN is granted read/write repository permission, even when it is triggered from a public fork. For more information, see "[Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target)."
```

https://docs.github.com/en/actions/using-workflows/reusing-workflows

```
If jobs.<job_id>.permissions is not specified in the calling job, the called workflow will have the default permissions for the GITHUB_TOKEN. For more information, see "[Automatic token authentication](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)."
The GITHUB_TOKEN permissions passed from the caller workflow can be only downgraded (not elevated) by the called workflow.
```

**Comment by kachick** [comment]
Could you share the actual used permissions from `Set up job` section in actual run?

My worked example is

```
GITHUB_TOKEN Permissions
  Actions: read
  Checks: read
  Contents: write
  Metadata: read
  PullRequests: write
```

**Comment by MarcHagen** [comment]
Yes, I've come to the same conclusion.

Minimal set of permissions needed, the missing `Contents: read` is enough for the GraphQL.
```
GITHUB_TOKEN Permissions
  Actions: read
  Checks: read
  Contents: read
  Metadata: read (always)
```

**Comment by MarcHagen** [comment]
Because the added check for workflow files for the `wait-list` and `skip-list` the content read is needed because of the fetching of files.
This seems a logical result of the new version, and explains why the older API didnt break and now it does "break".

What I'm missing was that the `content: read` at minimum is needed from 2.0 and forward.

Don't know if it is possible to only check content (workflow files) if the `skip-list` or `wait-list` are filled in.
So the `content: read` would not be required?? Not breaking for me but could be nice as a security feature.

**Comment by kachick** [comment]
Thanks for checking and sharing your permission, I got same result as v2 requires `contents: read` and v1 didn't require. Requiring `contents: write` is a wrong description in old documents. 🙇‍♂️ 

However I will go on current(v2) required permissions as requiring `contents: read`. The reasons are follows

- Permitting `contents: read` is still okay for me. I did think this action require `contents: write`. Just `read` is enough, it is not bad news.
- Removing the permission did not work even I removed to query for `workflow`
- Splitting or merging queries with given permissions should make complex implementation
- GraphQL looks like better for this purpose even we are not provide list-based mode, in REST(v3 API) era, I got own run id with another query.
- Getting workflow properties with one query helps our developing

So I just update the documents in #599
